libraries/AP_Motors/AP_MotorsHeli_Dual.cpp

```diff
@@ -516,17 +496,26 @@ void AP_MotorsHeli_Dual::move_actuators(float roll_out, float pitch_out, float c
     // ToDo: add main rotor cyclic power?
     _main_rotor.set_collective(fabsf(collective_out));
 
-    // compute swashplate tilt
-    float swash1_pitch = get_swashplate(1, AP_MOTORS_HELI_DUAL_SWASH_AXIS_PITCH, pitch_out, roll_out, yaw_out, collective_out_scaled);
-    float swash1_roll = get_swashplate(1, AP_MOTORS_HELI_DUAL_SWASH_AXIS_ROLL, pitch_out, roll_out, yaw_out, collective_out_scaled);
-    float swash1_coll = get_swashplate(1, AP_MOTORS_HELI_DUAL_SWASH_AXIS_COLL, pitch_out, roll_out, yaw_out, collective_out_scaled);
-    float swash2_pitch = get_swashplate(2, AP_MOTORS_HELI_DUAL_SWASH_AXIS_PITCH, pitch_out, roll_out, yaw_out, collective2_out_scaled);
-    float swash2_roll = get_swashplate(2, AP_MOTORS_HELI_DUAL_SWASH_AXIS_ROLL, pitch_out, roll_out, yaw_out, collective2_out_scaled);
-    float swash2_coll = get_swashplate(2, AP_MOTORS_HELI_DUAL_SWASH_AXIS_COLL, pitch_out, roll_out, yaw_out, collective2_out_scaled);
- 
-    // Calculate servo positions in swashplate library
-    _swashplate1.calculate(swash1_roll, swash1_pitch, swash1_coll);
-    _swashplate2.calculate(swash2_roll, swash2_pitch, swash2_coll);
+    // Mix swash plate
+    switch (_dual_mode) {
+        case AP_MOTORS_HELI_DUAL_MODE_TANDEM:
+            mix_tandem(pitch_out, roll_out, yaw_out, collective_out_scaled, collective2_out_scaled);
+            break;
+
+        case AP_MOTORS_HELI_DUAL_MODE_TRANSVERSE:
+            mix_transverse(pitch_out, roll_out, yaw_out, collective_out_scaled, collective2_out_scaled);
+            break;
+
+        case AP_MOTORS_HELI_DUAL_MODE_INTERMESHING:
+            mix_intermeshing(pitch_out, roll_out, yaw_out, collective_out_scaled, collective2_out_scaled);
+            break;
+
+        default:
```

Comment:
This is subtly different from the prior implementation where we effectively defaulted to tandem.
Might be worth adding a pre-arm on H_DUAL_MODE?  Just thinking of the user that has accidentally had it set to >3 and has previously had it working fine but now only has locked heads with nothing else to go on.  They will blame it on a firmware bug ....